File: FaceBook/Untagged/NestedIntegerString.java
 public String unfold(List<NestedInteger> nest) {
       String res = helper(nest,1);
        return res;
}
public String helper(List<NestedInteger> nest, int level) {
  if (nest.isEmpty()) return "";
  StringBuilder sb = new StringBuilder();
  for (NestedInteger item : nest) {
    if (item.isInteger()) {
      sb.append("+").append(item.getInteger());
    }
    else {
        String levelDepth = helper(item.getList(), level + 1);
        if (levelDepth.equals("")) continue;
      sb.append("+").append(levelDepth);
    }
  }
  sb.deleteCharAt(0);
  if (level == 1) return sb.toString();
  return sb.insert(0,"(").append(")").append("*").append(level).toString();
} 

File: FaceBook/Untagged/nested_integer_string.py
def to_formula(nest):
    return to_formula_helper(nest, 1)

def to_formula_helper(nest, depth):
    if not nest:
        return ''

    result = ''
    for i, ele in enumerate(nest):
        if ele.isInteger():
            result = result + str(ele.getInteger()) + '+'
        else:
            sub_result = to_formula_helper(ele.getList(), depth + 1)
            if sub_result:
                result = result + sub_result + '+'

    result.pop()
    if depth != 1:
        result = '(' + result + ')*'+ str(depth)

    return result
